### ISSUE TITLE
Move method implementations into own files

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/javascript-typescript-langserver#298]
TypeScriptService has grown in size over time. I would like to add a folder "features" that contains one file per feature with e.g. a "ReferenceProvider" class. The TypeScriptService would merely call these and map the returned Observables to JSON Patches.

**Comment by tomv564** [comment]
Would be a good opportunity to honour the `streaming` setting in `ClientCapabilities`! (Make the Observable consumption swappable between json-patch and the old behaviour)

**Comment by felixfbecker** [comment]
I don't think this refactor makes a difference for that - the current behaviour is to send both the partial results and the final result. Can't change that right now because our client implementation sometimes relies on the final result. The change would be only in connection.ts (there is a TODO comment).
